src/ducks/pullRequest/index.js

```diff
@@ -0,0 +1,59 @@
+/* @flow */
+import type { PullRequestGraphType } from 'services/ono/queries/pullRequest'
+export type { PullRequestGraphType } from 'services/ono/queries/pullRequest'
+
+export const types = {
+  FETCH_START: 'PULLREQUEST/FETCH_START',
+  FETCH_ERROR: 'PULLREQUEST/FETCH_ERROR',
+  FETCH_DONE: 'PULLREQUEST/FETCH_DONE',
+}
+
+const initialState = {
+  error: null,
+  isFetching: false,
+  pullRequest: null,
+}
+
+export type PullRequestStateType = {
+  error: ?string,
+  isFetching: boolean,
+  pullRequest: ?PullRequestGraphType,
+}
+
+export type PullRequestActionType =
+    { type: 'PULLREQUEST/FETCH_START' }
+  | { type: 'PULLREQUEST/FETCH_ERROR', error: string }
+  | { type: 'PULLREQUEST/FETCH_DONE', pullRequest: PullRequestGraphType }
+
+export default (
+  state: PullRequestStateType = initialState, action: PullRequestActionType
+) => {
+  switch (action.type) {
+    case types.FETCH_START:
+      return {
+        error: null,
+        isFetching: true,
+        pullRequest: null,
```

Comment:
Yes, one fetch saga for every simple flow. This was only meant for get `GET` case.

I would normally start simple and add more complexity/actions if needed. For API request I would usually just have 3 actions: `fetch`, `fail`, `success`. I'm basing this on:
http://redux.js.org/docs/advanced/AsyncActions.html

But it might make more sense to do what you propose... The way I see it, you also have 4 actions: `fetchStatus` (acts as two event, started and ended), `fail`, `success`. Which gives you a bit more granularity.

Do you have some links for me to read on the subject? I'll update the code.